azurerm/internal/services/storage/resource_arm_storage_account.go

```diff
@@ -1239,6 +1293,20 @@ func resourceArmStorageAccountRead(d *schema.ResourceData, meta interface{}) err
 		}
 	}
 
+	// FileStorage does not support blob settings
+	if resp.Kind != storage.BlobStorage && resp.Kind != storage.BlockBlobStorage && resp.Sku.Tier != storage.Premium {
```

Comment:
there's a crash here:

```suggestion
	if resp.Kind != storage.BlobStorage && resp.Kind != storage.BlockBlobStorage && resp.Sku != nil && resp.Sku.Tier != storage.Premium {
```